### PR TITLE
Python 3.11

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/channels/types/justcall/tests.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def test_claim(self):
         # try once with an error
         with patch("requests.post") as mock_post:
             mock_post.return_value = MockResponse(400, '{ "message": "Failed try again later." }')
-            with self.assertRaisesRegexp(ValidationError, "Unable to add webhook to JustCall: Failed try again later."):
+            with self.assertRaisesRegex(ValidationError, "Unable to add webhook to JustCall: Failed try again later."):
```
Comment (norkans7) [inline temba/channels/types/justcall/tests.py:35]: Removed in 3.12

File: temba/channels/types/justcall/tests.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def test_claim(self):
         # try once with an error
         with patch("requests.post") as mock_post:
             mock_post.return_value = MockResponse(400, '{ "message": "Failed try again later." }')
-            with self.assertRaisesRegexp(ValidationError, "Unable to add webhook to JustCall: Failed try again later."):
```
Comment (norkans7) [inline temba/channels/types/justcall/tests.py:35]: Removed in 3.12

File: pyproject.toml [hunk 3]
```diff
@@ -8,7 +8,7 @@ authors = ["Nyaruka <code@nyaruka.com>"]
 repository = "http://github.com/rapidpro/rapidpro"
 
 [tool.poetry.dependencies]
-python = "^3.10"
+python = "^3.11"
```
Comment (matlads) [inline pyproject.toml:11]: rapidpro seems to work just fine on python-3.10
```suggestion
python = ">=3.10<4"
```

Comment (rowanseymour) [reply pyproject.toml:11]: Currently yes but we'd like to start using newer Python features without worrying if they're supported on older versions so the goal is to get the required version to 3.12

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -4,8 +4,10 @@ jobs:
   test:
     name: Test
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.11.x", "3.12.x"]
```
Comment (rowanseymour) [inline .github/workflows/ci.yml:9]: really don't want to do double tests - just make this PR be 3.11 so we can get it out there - and then worry about 3.12 separately

File: pyproject.toml (hunk 3 above)
Comment (rowanseymour) [inline pyproject.toml:11]: @norkans7 in poetry `^3.11` means 3.11, 3.12, etc.. let's try `~3.11` to see if it installs with 3.11 even when 3.12 is available

File: pyproject.toml [hunk 5]
```diff
@@ -8,7 +8,7 @@ authors = ["Nyaruka <code@nyaruka.com>"]
 repository = "http://github.com/rapidpro/rapidpro"
 
 [tool.poetry.dependencies]
-python = "^3.10"
+python = "^3.11.9"
```
Comment (rowanseymour) [inline pyproject.toml:11]: `~3.11`

**Review by rowanseymour — CHANGES_REQUESTED** [review]

**Pull request merged**